servicecontrol/servicecontrol-in-practice.md

```diff
@@ -0,0 +1,27 @@
+title: ServiceControl in Practice
+summary: Details how to think about the use of ServiceControl in production environments
+tags:
+- ServiceControl
+---
+
+ServiceControl is a tool that can accommodate many situations. Certain features come with a cost and in some cases the cost of having a feature active is not worth the performance impact on the system.
+
+For example, in your development environment you will want a lot of logging information to support problem analysis. For many users the [Debug Session](/servicecontrol/plugins/debug-session.md) plugin is really useful at this stage of the application life-cycle but you would never put that plugin into production. For the same reason, we advise that customers never use the [Saga Audit](/servicecontrol/plugins/saga-audit.md) plugin outside of development.
```

Comment:
The Debug Session plugin is a bad example. It existed to serve ServiceMatrix. All it does is paint a debug session id header on outgoing messages. ServiceInsight does not currently use this info. I would state that the Saga Audit plugin is useful in a development context but generates a lot of data and slows down the endpoint containing the saga. This is OK under low load in a development environment but will cause memory, cpu, and disk pressure under high load (like performance testing and production environments).